tensorboard/webapp/runs/views/runs_table/regex_edit_dialog_container.ts

```diff
@@ -60,17 +64,34 @@ const INPUT_CHANGE_DEBOUNCE_INTERVAL_MS = 500;
 })
 export class RegexEditDialogContainer {
   private readonly experimentIds: string[];
+  private readonly expNameByExpId: Record<string, string>;
   private readonly runIdToEid$: Observable<Record<string, string>>;
   private readonly allRuns$: Observable<Run[]>;
   private readonly tentativeRegexString$: Subject<string> =
     new Subject<string>();
+  private readonly tentativeRegexType$: Subject<GroupByKey> =
+    new Subject<GroupByKey>();
 
   readonly groupByRegexString$: Observable<string> = defer(() => {
     return merge(
       this.store.select(getColorGroupRegexString).pipe(take(1)),
       this.tentativeRegexString$
     );
-  }).pipe(startWith(''));
+  }).pipe(startWith(''), shareReplay(1));
+
+  readonly groupByRegexType$: Observable<GroupByKey> = merge(
+    this.store.select(getRunGroupBy).pipe(
+      take(1),
+      map((group) => group.key)
+    ),
+    this.tentativeRegexType$
+  ).pipe(
+    filter(
+      (key) => key === GroupByKey.REGEX || key === GroupByKey.REGEX_BY_EXP
+    ),
+    startWith(GroupByKey.REGEX),
```

Comment:
Do we need the startWith here? IIUC this observable should always start with the GroupByKey of the store anyway

Reply:
Just checked, seems like we don't need.

Reply:
Seems this hasn't been removed yet

Reply:
Oops. Just removed.

Reply:
Update: It is still needed, as we filter this observable by regex or regex_by_exp and if user haven't touched them, this observable will stall everything else.

Reply:
You're right, GroupByKey will initially start at something else like RUNS. Thanks for the correction!